### DISCUSSION TITLE
Does Cypress sometimes break Angular change detection?

### DISCUSSION THREAD

**Discussion by diminutivesloop** [cypress-io/cypress#15147] [Questions and Help]
I was writing a test using the `cy.realHover` command provided by [cypress-real-events](https://github.com/dmtrKovalenko/cypress-real-events) to verify that options in a menu component are highlighted when mousing over them. However when the tests failed I noticed that the option highlighting wasn't working even when I tried it manually in the test window. However when I open the test harness page (a Storybook iframe page in this case) the highlighting works as expected. The component is built with Angular and the highlighting is triggered via a `mouseover` event binding coupled with a `class` binding that applies a css class for highlighted options. From my testing it appears that the `mouseover` event is firing but the `class` binding isn't being updated. I'm probably just going to put this test aside since it's not critical but I'm curious if there's a known issue with Cypress interfering with Angular change detection.

**Comment by portyu9** [comment]
Hi @diminutivesloop 

For this component you don't need `cy.realHover()`.

You are testing an Angular `(mouseover)` event binding, so trigger that event directly:

```js
cy.get('[data-cy="menu-option"]')
  .trigger('mouseover')
  .should('have.class', 'highlighted')
```

`cypress-real-events` is useful when you specifically need a native browser hover—for example CSS `:hover`—but it adds another event-delivery mechanism that isn't necessary for an Angular event handler.

So I would first replace `realHover()` with Cypress's `.trigger('mouseover')`. Angular should then process the event and run its normal change-detection cycle.

Please mark this answer as accepted if it helped.
